func/mpi/mpi_cartesian.cpp

```diff
@@ -14,14 +15,23 @@ int main(int argc, char* argv[])
     MPI_Comm_size(MPI_COMM_WORLD, &worldSize);
 
     // Prepare arguments
-    int ndims = 2;
-    int sideLength = static_cast<int>(std::floor(std::sqrt(worldSize)));
-    int nprocs = sideLength * sideLength;
-    int dims[2] = { sideLength, sideLength };
-    int periods[2] = { 0, 0 };
+    // Note that the grid size is always 5x1x1 because the tests are always
+    // ran with MPI_WORLD_SIZE = 5
```

Comment:
We shouldn't assume this will always be the case. Add a check in this function that will error if it's not 5

Reply:
Sure, I also lurk your PRs sometimes :detective: (plus nice new word).